spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/InstanceType.java

```diff
@@ -0,0 +1,26 @@
+package org.apache.spark.metrics.sink.statsd;
+
+public enum InstanceType {
+    DRIVER("driver"),
+    EXECUTOR("executor"),
+    UNDEFINED("undefined");
+
+    private String value;
+
+    InstanceType(String value) {
+        this.value = value;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public static InstanceType fromString(String value) {
+        for (InstanceType instanceType : InstanceType.values()) {
+            if (instanceType.value.equalsIgnoreCase(value)) {
+                return instanceType;
+            }
+        }
+        throw new IllegalArgumentException("No constant with value " + value + " found");
```

Comment:
Good catch, thanks. So the only place where this method is called is in `InstanceDetailsProvider`:
```
InstanceType.fromString(SparkEnv.get().metricsSystem().instance())
```

In this specific use case (Spark on Mesos) we will see only two values: `driver` and `executor` because Dispatcher doesn't have `SparkEnv`. The exception here is to address some unexpected value and react to it.

Given that, I think the best way would be to use non-parameterized enum and `valueOf` which will throw `IllegalArgumentException` in case the value is not found.